Index: sources/org/apache/batik/bridge/SVGTextElementBridge.java
===================================================================
--- sources/org/apache/batik/bridge/SVGTextElementBridge.java	(Revision 805929)
+++ sources/org/apache/batik/bridge/SVGTextElementBridge.java	(Arbeitskopie)
@@ -550,6 +550,15 @@
         }
     }
 
+    public void handleDOMAttrModifiedEvent(MutationEvent evt) {
+        Node childNode = (Node)evt.getTarget();
+        //if the parent is displayed, then discard the layout.
+        if (isParentDisplayed(childNode)) {
+            laidoutText = null;
+            computeLaidoutText(ctx, e, getTextNode());
+        }
+    }
+
     /**
      * Invoked when an MutationEvent of type 'DOMNodeRemoved' is fired.
      */
@@ -692,7 +701,8 @@
                     || ln.equals(SVG_DY_ATTRIBUTE)
                     || ln.equals(SVG_ROTATE_ATTRIBUTE)
                     || ln.equals(SVG_TEXT_LENGTH_ATTRIBUTE)
-                    || ln.equals(SVG_LENGTH_ADJUST_ATTRIBUTE)) {
+                    || ln.equals(SVG_LENGTH_ADJUST_ATTRIBUTE)
+                    || ln.equals(SVG_START_OFFSET_ATTRIBUTE)) {
                 char c = ln.charAt(0);
                 if (c == 'x' || c == 'y') {
                     getTextNode().setLocation(getLocation(ctx, e));
@@ -2157,7 +2167,7 @@
          * Invoked when an MutationEvent of type 'DOMAttrModified' is fired.
          */
         public void handleDOMAttrModifiedEvent(MutationEvent evt) {
-            //nothing to do
+            textBridge.handleDOMAttrModifiedEvent(evt); 
         }
 
         /**
@@ -2194,6 +2204,7 @@
          */
         public void handleAnimatedAttributeChanged
                 (AnimatedLiveAttributeValue alav) {
+	    textBridge.handleAnimatedAttributeChanged(alav);
         }
 
         /**
Index: sources/org/apache/batik/bridge/SVGTextPathElementBridge.java
===================================================================
--- sources/org/apache/batik/bridge/SVGTextPathElementBridge.java	(Revision 805929)
+++ sources/org/apache/batik/bridge/SVGTextPathElementBridge.java	(Arbeitskopie)
@@ -23,6 +23,7 @@
 import java.awt.geom.GeneralPath;
 
 import org.apache.batik.dom.util.XLinkSupport;
+import org.apache.batik.dom.svg.SVGOMTextPathElement;
 import org.apache.batik.gvt.text.TextPath;
 import org.apache.batik.parser.AWTPathProducer;
 import org.apache.batik.parser.ParseException;
@@ -114,7 +115,7 @@
         TextPath textPath = new TextPath(new GeneralPath(pathShape));
 
         // set the start offset if specified
-        s = textPathElement.getAttributeNS(null, SVG_START_OFFSET_ATTRIBUTE);
+        s = ((SVGOMTextPathElement)textPathElement).getStartOffset().getAnimVal().getValueAsString();
         if (s.length() > 0) {
             float startOffset = 0;
             int percentIndex = s.indexOf('%');
